rig/bitfield.py

```diff
@@ -137,12 +148,27 @@ def add_field(self, identifier, length=None, start_at=None, tags=None):
 
         # Check for fields which occupy the same bits
         if start_at is not None:
-            end_at = start_at + (length or 1)
+            if relative_to_lsb:
+                lsb_start_at = start_at
+                end_at = start_at + (length or 1)
+            else:
+                # Switch fields relative to the MSB to be equivalent to fields
+                # relative to the LSB
+                lsb_start_at = (self.length - start_at - (length or 1))
+                end_at = (self.length - start_at)
+
             for other_identifier, other_field in self._potential_fields():
                 if other_field.start_at is not None:
-                    other_start_at = other_field.start_at
-                    other_end_at = other_start_at + (other_field.length or 1)
-                    if end_at > other_start_at and other_end_at > start_at:
+                    if other_field.relative_to_lsb:
+                        other_start_at = other_field.start_at
+                        other_end_at = \
+                            other_start_at + (other_field.length or 1)
+                    else:
+                        other_start_at = (self.length - other_field.start_at -
+                                          (other_field.length or 1))
+                        other_end_at = (self.length - other_field.start_at)
```

Comment:
Hmm... this will break if the field we're comparing against has already been assigned.  When assigning a field the "relative_to_lsb" tag should be set to true.